snapcraft/internal/common.py

```diff
@@ -255,3 +255,39 @@ def format_output_in_columns(elements_list, max_width=MAX_CHARACTERS_WRAP,
         result_output.append(sep.join(candidate_output[i]))
 
     return result_output
+
+
+def get_include_paths(root, arch_triplet):
+    paths = [
+        os.path.join(root, 'include'),
+        os.path.join(root, 'usr', 'include'),
+        os.path.join(root, 'include', arch_triplet),
+        os.path.join(root, 'usr', 'include', arch_triplet),
+    ]
+
+    return [p for p in paths if os.path.exists(p)]
+
+
+def get_library_paths(root, arch_triplet):
+    paths = [
+        os.path.join(root, 'lib'),
+        os.path.join(root, 'usr', 'lib'),
+        os.path.join(root, 'lib', arch_triplet),
+        os.path.join(root, 'usr', 'lib', arch_triplet),
+    ]
+
+    return [p for p in paths if os.path.exists(p)]
+
+
+def combine_paths(paths, prepend, separator):
+    paths = ['{}{}'.format(prepend, p) for p in paths]
+    return separator.join(paths)
+
+
+def format_path_variable(envvar, paths, prepend, separator):
+    if not paths:
+        return ''
```

Comment:
So in the uses of `format_path_variable` I see you always check for `path`, so instead wouldn't it be better here to raise a `ValueError`?